test/readme.md

```diff
@@ -143,6 +144,18 @@ modify the steps to work with existing Node projects.
 
 > **Note:** When making driver changes, you will need to run `npm run build:ts` with each change in order for it to take effect.
 
+## Writing Tests
+
+> TODO: flesh this section out more
+
+We use mocha to construct our test suites and chai to assert expectations.
+
+Some special notes on how mocha works with our testing setup:
+
+- `before` hooks will run even if a test is skipped by the environment it runs on.
+  - So, for example, if your before hook does logic that can only run on a certain server version you can't depend on your test block metadata to filter for that.
+- `after` hooks cannot be used to clean up clients because the session leak checker currently runs its afterEach hook first.
```

Comment:
```suggestion
- `after` hooks cannot be used to clean up clients because the session leak checker currently runs in an `afterEach` hook, which would be executed before any `after` hook has a chance to run
```